tools/parse/src/main/scala/org/allenai/nlpstack/parse/poly/ml/LabeledFeatureVectors.scala

```diff
@@ -15,19 +16,29 @@ object FeatureEncoding {
   implicit val jsFormat = jsonFormat1(FeatureEncoding.apply)
 }
 
+/** Support structure to store a feature vector per parse family with associated label for expected
+  * outcome.
+  */
+case class LabeledFeatureVectorPerParseFamily(
+    sentence: String, family: LabeledFamily, featureVector: FeatureVector, expectedOutcome: Int
+) {
+}
+
 /** Abstraction for a set of labeled feature vectors.
   *
   * Provides various serialization options for different machine learning tools.
   *
-  * @param labeledVectors a sequence of feature vectors labeled with integer outcomes
+  * @param labeledVectors a sequence of families and their feature vectors labeled with
+  * integer outcomes
   */
-case class TrainingData(labeledVectors: Iterable[(FeatureVector, Int)]) {
+case class LabeledFeatureVectors(
+    labeledVectors: Iterable[LabeledFeatureVectorPerParseFamily]
```

Comment:
Ah, so an issue with this change is that it makes the ML package non-general purpose. As a rule of thumb, the ML sublibrary should not depend on anything from either the polyparser or reranking subpackages, since it's intended to be a general-purpose classification library.